moveit_planners/pilz_industrial_motion_planner/include/pilz_industrial_motion_planner/joint_limits_container.h

```diff
@@ -144,10 +144,11 @@ class JointLimitsContainer
    * @brief verify position limits of multiple joints
    * @param joint_names
    * @param joint_positions
+   * @param first_failing_joint returns first joint that fails position limits (if any)
    * @return
    */
-  bool verifyPositionLimits(const std::vector<std::string>& joint_names,
-                            const std::vector<double>& joint_positions) const;
+  bool verifyPositionLimits(const std::vector<std::string>& joint_names, const std::vector<double>& joint_positions,
+                            std::string* first_failing_joint = nullptr) const;
```

Comment:
I don't understand why you restrict the API to report only the first offending joint. If there are multiple joints, this will require a tedious iteration to fix all of them.
I suggest passing a reference to a `vector<string>` instead.  If your main motivation was API compatibility, you could directly fill the (optional) string with a comma-separated list of joint names.

Reply:
my main motivation was to avoid iterating over everything when the original code did not. but I can also change that. :shrug:

Reply:
In normal operation, i.e. without joint violations, the code needs to iterate all joints anyway and the failure case isn't time-critical, is it?